metadeploy/api/jobs.py

```diff
@@ -283,26 +284,22 @@ def create_scratch_org(*, plan_id, org_name, result_id):
         )
         async_to_sync(preflight_started)(org, preflight_result)
         preflight(preflight_result.pk)
-
-    # @@@ TODO: start installation job automatically if:
-    #   - Plan has no preflight
-    #   - Plan has no optional steps
-    # job = Job.objects.create(
-    #     user=None,
-    #     plan=plan,
-    #     is_public=True,
-    #     org_id=scratch_org_config.config["org_id"],
-    # )
-
-    # rq_job = run_flows.delay(
-    #     plan=plan,
-    #     skip_steps=[],
-    #     result_class=Job,
-    #     result_id=job.id,
-    # )
-    # job.job_id = rq_job.id
-    # job.enqueued_at = rq_job.enqueued_at
-    # job.save()
+    elif plan.required_step_ids.count() == plan.steps.count():
+        # Start installation job automatically if both:
+        # - Plan has no preflight
+        # - All plan steps are required
+        with transaction.atomic():
```

Comment:
@duggiemitchell It means that the changes within this block are committed to the database all at once, if the entire block succeeds (in this case, so that the job steps are saved along with the new job itself). https://docs.djangoproject.com/en/3.0/topics/db/transactions/#django.db.transaction.atomic